dagmc_stats/dagmc_stats.py

```diff
@@ -460,23 +463,85 @@ def get_lri(vert_i, gc_all, tri_vert_data, my_core):
     return Lri
 
 
-def get_roughness(my_core, native_ranges):
+def get_roughness(my_core, native_ranges, tag=False):
     """Get local roughness values of all the non-isolated vertices
 
     inputs
     ------
     my_core : a MOAB Core instance
     native_ranges : a dictionary containing ranges for each native type
     in the file (VERTEX, TRIANGLE, ENTITYSET)
+    tag : whether or not to add the roughness tag
 
     outputs
     -------
-    roughness : (numpy array) the roughness for all surfaces in the meshset
+    roughness : (dictionary) the roughness for all vertices in the meshset
+    stored in the form of vert : local roughness value
     """
     tri_vert_data, all_verts = get_tri_vert_data(my_core,
                                                     native_ranges[types.MBTRI])
     gc_all = get_gaussian_curvature(my_core, all_verts, tri_vert_data)
-    roughness = np.zeros(len(all_verts))
-    for vert_idx, vert_i in enumerate(all_verts):
-        roughness[vert_idx] = get_lri(vert_i, gc_all, tri_vert_data, my_core)
+    roughness = {}
+    for vert_i in all_verts:
+        roughness[vert_i] = get_lri(vert_i, gc_all, tri_vert_data, my_core)
+
+    if tag:
+        #add triangle average roughness tag
+        lr_tag = {}
+        for tri in native_ranges[types.MBTRI]:
+            three_verts = list(my_core.get_adjacencies(tri, 0, op_type=1))
+            sum_lr = 0
+            for vert in three_verts:
+                sum_lr += roughness[vert]
+            lr_tag[tri] = sum_lr/3
+        add_tag(my_core, 'FACET_AVG_LR', lr_tag, types.MB_TYPE_DOUBLE)
```

Comment:
Separation of concerns: Don't call `add_tag` from within this method.  Allow the user to calculate the roughness of vertices and roughness of triangles without changing the mesh DB (adding a tag changes the mesh DB).  By saving this data, the user can then call `add_tag` separately in the method that calls this.